src/TLuaInterpreter.cpp

```diff
@@ -8243,9 +8243,9 @@ int TLuaInterpreter::permKey(lua_State* L)
 
     Host& host = getHostFromLua(L);
     TLuaInterpreter* pLuaInterpreter = host.getLuaInterpreter();
-    auto validationResult = pLuaInterpreter->validLuaCode(luaFunction);
-    if(!validationResult.first) {
-        lua_pushfstring(L, "permKey: bad argument #%d (invalid Lua code: %s)", argIndex, validationResult.second.toUtf8().constData());
+    auto [validationResult, validationMessage] = pLuaInterpreter->validLuaCode(luaFunction);
+    if (!validationResult) {
+        lua_pushfstring(L, "permKey: bad argument #%d (invalid Lua code: %s)", argIndex, validationMessage.toUtf8().constData());
         return lua_error(L);
```

Comment:
Forgot about this earlier, but you can also combine it all into one if:

```suggestion
    if (auto [successful, errorMessage = pLuaInterpreter->validLuaCode(luaFunction); !successful) {
        lua_pushfstring(L, "permKey: bad argument #%d (invalid Lua code: %s)", argIndex, errorMessage.toUtf8().constData());
        return lua_error(L);
```

It's another cool feature of modern C++ that we use ([background](https://skebanga.github.io/if-with-initializer/)).

Reply:
Oh really cool :)